src/qnngds/cells.py

```diff
@@ -334,9 +336,14 @@ def nanowires(
     NANOWIRES = Device(f"NWIRES({cell_text})")
     nanowires_ref = []
     for i, channel_source_w in enumerate(channels_sources_w):
-        nanowire_ref = NANOWIRES << device.nanowire.spot(
-            channel_source_w[0], channel_source_w[1]
-        )
+        if lengths is None:
```

Comment:
probably want to uncomment the else?

Reply:
Yes sorry -- before I had things set up correctly my code couldn't "see" the variable_lengths function so I was unable to test this part of the code, and I commented it out to be safe. Currently working on testing the else and adding a couple other improvements

Reply:
sounds good, let me know when the PR is ready